library/src/main/java/com/schibsted/spain/barista/internal/matcher/TextColorMatcher.kt

```diff
@@ -1,43 +1,57 @@
+import android.util.TypedValue
 import android.view.View
 import android.widget.TextView
+import androidx.core.content.ContextCompat
+import androidx.test.espresso.matcher.BoundedMatcher
+import com.schibsted.spain.barista.internal.util.ColorResourceType
+import com.schibsted.spain.barista.internal.util.colorResourceType
 import org.hamcrest.Description
 
-class TextColorMatcher(@ColorRes private val colorRes: Int) : BoundedMatcher<View, TextView>(TextView::class.java) {
+class TextColorMatcher(private val expectedColor: Int) : BoundedMatcher<View, TextView>(TextView::class.java) {
 
   private var colorName: String? = null
 
   override fun matchesSafely(textView: TextView): Boolean {
-    val context = textView.context
-    val resources = context.resources
-    colorName = resources.getResourceEntryName(colorRes)
 
-    val matchesColor = matchesColor(context, textView)
-    val matchesColorStateList = matchesColorList(context, textView)
+    colorName = when(expectedColor.colorResourceType) {
+      ColorResourceType.COLOR_RES -> textView.context.resources.getResourceEntryName(expectedColor)
+      ColorResourceType.COLOR_ATTR -> textView.context.resources.getResourceEntryName(expectedColor)
+      ColorResourceType.COLOR_INT -> "$expectedColor"
+    }
 
-    return matchesColor || matchesColorStateList
+    return matchesColor(textView) || (expectedColor.colorResourceType == ColorResourceType.COLOR_RES && matchesColorList(textView))
   }
 
-  private fun matchesColor(context: Context, textView: TextView): Boolean {
+  private fun matchesColor(textView: TextView): Boolean {
     val currentColorInt = textView.currentTextColor
-    val expectedColorInt = ContextCompat.getColor(context, colorRes)
+
+    val expectedColorInt = when (expectedColor.colorResourceType) {
+      ColorResourceType.COLOR_RES -> ContextCompat.getColor(textView.context, expectedColor)
+      ColorResourceType.COLOR_INT -> expectedColor
+      ColorResourceType.COLOR_ATTR -> textView.context.getColorAttribute(expectedColor)
+    }
+
     return currentColorInt == expectedColorInt
   }
 
-  private fun matchesColorList(context: Context, textView: TextView): Boolean {
+  private fun Context.getColorAttribute(colorAttr: Int): Int {
+    val outValue = TypedValue()
+    theme.resolveAttribute(colorAttr, outValue, true)
+    return outValue.data
+  }
+
+  private fun matchesColorList(textView: TextView): Boolean {
     val currentColorList = textView.textColors
-    val expectedColorList = ContextCompat.getColorStateList(context, colorRes)
-    return currentColorList == expectedColorList
+    val expectedColorInt = ContextCompat.getColorStateList(textView.context, expectedColor)
+    return currentColorList == expectedColorInt
   }
 
   override fun describeTo(description: Description) {
     if (colorName == null) {
-      description.appendText("with text color resource: [$colorRes]")
+      description.appendText("with text color: [$expectedColor]")
```

Comment:
Do you think we could give different messages depending on the received color? It would be great for the user I think.

Reply:
Done!

<img width="935" alt="Captura de pantalla 2020-04-09 a las 13 46 09" src="https://user-images.githubusercontent.com/887462/78891836-8aacf680-7a68-11ea-9506-c9d0dfe8c6d4.png">

<img width="974" alt="Captura de pantalla 2020-04-09 a las 13 46 53" src="https://user-images.githubusercontent.com/887462/78891857-939dc800-7a68-11ea-82c3-ca73f2a762fa.png">


<img width="972" alt="Captura de pantalla 2020-04-09 a las 13 47 29" src="https://user-images.githubusercontent.com/887462/78891907-a87a5b80-7a68-11ea-8653-d53b5a4dcf2a.png">